Add Emulated Kasa (#14391)
home-assistant/home-assistant.github.io: source/_integrations/emulated_kasa.markdown

---
title: Emulated Kasa
description: Instructions on how to integrate Emulated Kasa within Home Assistant.
ha_category:

>   - Hub

Reviewer: This by itself is not a Hub.

```suggestion
```

  - Energy
ha_iot_class: Local Push
ha_release: 0.112
ha_config_flow: false
ha_codeowners:
  - '@kbickar'
ha_domain: emulated_kasa
---

The Emulated Kasa integration emulates a TP-Link Kasa smart plug and announces the power usage of configured devices to any that might request it on the local network. 

> For example, the [Sense Energy Monitor](/integrations/sense) can use this to idenfity power usage.

Reviewer: ```suggestion
For example, the [Sense Energy Monitor](/integrations/sense) can use this to identify power usage.
```

The configuration includes a list of entities to expose with attributes for the published name and current power usage.
If the entity is a sensor or has a `current_power_w` attribute (such as in a smart switch), that value will be reported as the current power usage unless the power field is defined.

> The power field can contain a hard coded value, a sensor, or a template (see configuration example). 

Reviewer: ```suggestion
The power field can contain a hardcoded value, a sensor, or a template (see configuration example).
```

> <div class='note'>
> The provided power unit must be the current power usage in Watts.  Values of `kw` can be converted, but values of `kwh` cannot be used.
> </div>

Reviewer: ```suggestion
<div class='note'>

The provided power unit must be the current power usage in Watts.  Values of `kW` can be converted, but the values of `kWh` cannot be used.

</div>
```

> 

Reviewer: ```suggestion
```

## Configuration

This integration requires the entities exposed to be listed in your `configuration.yaml` file:

```yaml
# Example configuration.yaml entry
emulated_kasa:
  entities:
    light.dining_room:
      name: "Dining Room Lights"

>       power: 40.2

Reviewer: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

```

{% configuration %}
entities:

>   description: A list of entities exposed

Reviewer: ```suggestion
  description: A list of entities exposed.
```

  required: true
  type: map
  keys:
    name:

>       description: Name visible to external devices

Reviewer: ```suggestion
      description: Name visible to external devices.
```

      required: false
      type: string
    power:

>       description: The current power usage in Watts.  This can be set to a static value or a template 

Reviewer: ```suggestion
      description: The current power usage in watts. This can be set to a static value or a template.
```

      required: false
      type: [integer,float,template]
    power_entity:

>       description: A sensor measuring the current power usage in watts 

Reviewer: ```suggestion
      description: A sensor measuring the current power usage in watts.
```
